src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -91,15 +104,27 @@ private static VersionInfo GetVersionInfo()
             loaderAssemblyName: typeof(OnnxTransform).Assembly.FullName);
         }
 
-        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile)
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, int gpuDeviceId = -1, bool fallbackToCpu = false)
         {
-            var args = new Arguments { ModelFile = modelFile, InputColumns = new string[] { }, OutputColumns = new string[] { } };
+            var args = new Arguments {
+                ModelFile = modelFile,
+                InputColumns = new string[] { },
+                OutputColumns = new string[] { },
+                GpuDeviceId = gpuDeviceId,
+                FallbackToCpu = fallbackToCpu };
+
             return Create(env, args, input);
         }
 
-        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, string[] inputColumns, string[] outputColumns)
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, string[] inputColumns, string[] outputColumns, int gpuDeviceId = -1, bool fallbackToCpu = false)
```

Comment:
>public [](start = 8, length = 6)

can all Create method be made non-public, preferably private?

Reply:
I believe the tendency is to use the constructors to instantiate, and leave the static Create methods for the reflection-based component catalog approach if you don't have a private constructor with just environment and arguments (matching the signature. )

Wondering if you even need the Create methods. 

---
In reply to: [243467291](https://github.com/dotnet/machinelearning/pull/1922#discussion_r243467291) [](ancestors = 243467291)